src/big_mess/agg_classification_eval.py

```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
+"""agg_classification_eval.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1oxlQ0p57z8f8PTVuej3ifeFouvITjHxQ
+"""
+
+# !pip install datashader
+# !pip install holoviews hvplot colorcet
+# !pip install geoviews
+# !pip install pyspark
+
+from typing import Tuple, List
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+import sklearn
+from sklearn.metrics import confusion_matrix, accuracy_score, precision_score, recall_score, f1_score
+import holoviews as hv
+import geoviews as gv
+import geoviews.tile_sources as gts
+from holoviews.operation.datashader import datashade
+import matplotlib as mpl
+from pyspark.sql import DataFrame as SparkDataFrame, Window
+
+def overYearMonthStats(df: SparkDataFrame, columns: List[str], n: int) -> SparkDataFrame:
+    """
+    Funkcja licząca średnie dla wybranych zmiennych dla danych współrzędnych geograficznych
+    dla każdego z 12 miesięcy od roku w rozważanym pomiarze do n lat wstecz (tj. gdyby n=3, to
+    średnia zostałaby wyliczona z rozważanego roku oraz 3 lat przed nim, czyli w praktyce mielibyśmy
+    średnią z łącznie 4 lat).
+
+    :param df: zbiór, na którym chcemy przeprowadzać operacje
+    :param columns: lista nazw rozważanych kolumn ze zbioru danych
+    :param n: liczba lat wstecz, z których chcemy wyliczyć średnią
+    """
+
+    if not columns:
+      raise ValueError("Podana musi zostać co najmniej jedna zmienna")
```

Comment:
To już trochę poboczne, ale value error oznacza, że trafia nam się jakaś wartość w runtime której nie powinniśmy otrzymać nigdy. Coś jak NaN na wejściu funkcji. Do kontroli parametru odpowiedniejszy byłby assert.

Możesz @adadomagala1 zamknąć uwagę po poprawieniu